### PR TITLE
Split matrix server selection into utility function

### PR DESCRIPTION
**Pull request by andrevmatos**
Follow-up of #3393 , implement some leftovers from that PR

### CODE REVIEW

File: raiden/network/transport/matrix/utils.py [hunk 1]
```diff
@@ -234,3 +236,71 @@ def validate_userid_signature(user: User) -> Optional[Address]:
...
+def sort_servers_closest(servers: Sequence[str]) -> Sequence[Tuple[str, float]]:
+    """Sorts a list of servers by http round-trip time
+
+    Params:
+        servers: sequence of http server urls
+    Returns:
+        sequence of pairs of url,rtt in seconds, sorted by rtt, excluding failed servers
+        (possibly empty)
+    """
+    if not {urlparse(url).scheme for url in servers} <= {'http', 'https'}:
```
Comment (palango) [inline raiden/network/transport/matrix/utils.py]: can we call `issubset` here instead of `<=`? It took me some time to understand this.

**Comment by palango** [comment]
I also don't see the `send_global` function. Did you forget to push that?

File: raiden/network/transport/matrix/utils.py (hunk 1 above)
Comment (andrevmatos) [reply raiden/network/transport/matrix/utils.py]: Sure, they're equivalent, but will do for readability

**Comment by andrevmatos** [comment]
@palango the PR was still in WIP state, I would still add it. But as this was already getting a little big, decided to split the PR and do this only as the promised follow-up of #3393 . The `send_global`/MS PR will follow shortly

File: raiden/network/transport/matrix/transport.py [hunk 2]
```diff
@@ -1268,18 +1224,20 @@ def _get_room_ids_for_address(
...
 
         **MUST** be called from a context that holds the `_account_data_lock`.
-        _address_to_room_ids: Dict[AddressHex, List[_RoomID]]
         """
         _msg = '_leave_unused_rooms called without account data lock'
         assert self._account_data_lock.locked(), _msg
 
-        # TODO: Remove the next two lines and check if transfers start hanging again
-        self._client.set_account_data('network.raiden.rooms', _address_to_room_ids)
+        # TODO: Remove the next five lines and check if transfers start hanging again
```
Comment (rakanalh) [inline raiden/network/transport/matrix/transport.py]: The next four i would say, no?

File: raiden/network/transport/matrix/transport.py [hunk 3]
```diff
@@ -273,40 +274,29 @@ def __init__(self, config: dict):
...
         self._raiden_service: RaidenService = None
 
+        if config['server'] == 'auto':
+            available_servers = config['available_servers']
+        elif urlparse(config['server']).scheme in {'http', 'https'}:
+            available_servers = [config['server']]
+        else:
+            raise TransportError('Invalid matrix server specified (valid values: "auto" or a URL)')
+
         def _http_retry_delay() -> Iterable[float]:
```
Comment (ulope) [inline raiden/network/transport/matrix/transport.py]: Shouldn’t this be part of the utility?

File: raiden/network/transport/matrix/transport.py (hunk 2 above)
Comment (ulope) [reply raiden/network/transport/matrix/transport.py]: The `return` counts as well ;)

File: raiden/network/transport/matrix/transport.py (hunk 3 above)
Comment (andrevmatos) [reply raiden/network/transport/matrix/transport.py]: @ulope this had too much to do with internals and variables of `config`, and I wanted to make the utility function as agnostic in relation to config as possible, that's why I kept this outside of it, but can be as easily used outside raiden as any other utility function, so we can say it's _already split_

**Pull request merged**